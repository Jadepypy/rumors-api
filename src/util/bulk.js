const BATCH_SIZE = 100;

export default class Bulk {
  constructor(client, batchSize = BATCH_SIZE) {
    this.client = client;
    this.batchSize = batchSize;
    this._operations = [];
    this.actionsCount = 0;
  }

  async push(items, count = 1) {
    this._operations.push(...items);
    this.actionsCount += count;
    if (
      this.actionCounts > this.batchSize ||
      this._operations.length > 3 * this.batchSize
    ) {
      await this.flush();
    }
    return this;
  }

  async flush() {
    if (this._operations.length === 0) {
      return this;
    }

    // Process entires queued up
    const op = this._operations.splice(0);
    this.actionCounts = 0;
    const { body } = await this.client.bulk({
      body: op,
      refresh: 'true',
    });
    if (body.errors) {
      console.error(body.errors);
    } else {
      console.info(`${body.items.length} items processed`);
    }

    return this;
  }
}
