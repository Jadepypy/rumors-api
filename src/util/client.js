import elasticsearch from '@elastic/elasticsearch';

export default new elasticsearch.Client({
  node: process.env.ELASTICSEARCH_URL,
});

// Processes {_id, _version, found, _source: {...}} to
// {id, ..._source}.
//
export function processMeta({
  _id: id,
  _source: source,

  found, // for mget queries
  _score, // for search queries

  sort, // cursor when sorted
}) {
  if (found || _score !== undefined) {
    return { id, ...source, _cursor: sort, _score };
  }
  return null; // not found
}
