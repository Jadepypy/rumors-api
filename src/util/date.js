const dateFormat = /^[0-9]{4}-[0-9]{2}-[0-9]{2}$/;

/**
 * Given a date string, determine if it's a valid date, and return a Date object if it is.

 * @param {string} dateStr        A string representation of a date in YYYY-MM-DD
 * @param {string} [name='Date']  A string used in error message

 * @return {Date}
 */
const convertAndValidateDate = (dateStr, name = 'Date') => {
  if (!dateStr.match(dateFormat)) {
    throw new Error(`${name} must be in the format of YYYY-MM-DD`);
  }
  try {
    const date = new Date(dateStr);
    if (date.toString() !== 'Invalid Date') {
      return date;
    } else {
      throw new Error();
    }
  } catch {
    throw new Error(`${name} must be a valid date in the format of YYYY-MM-DD`);
  }
};

/**
 * Given two date strings and optional maxDuration, determine if it's a valid
   date range.

 * @param {string} startDateStr             A string representation of start date in YYYY-MM-DD.
 * @param {string} endDateStr               A string representation of end date in YYYY-MM-DD.
 * @param {number} [maxDuration]            Optional maximun duration in milliseconds.
   @param {bool}   [allowFutureDate=false]  Whether the date range allows end dates in the future.
 * @return {object}                         Validation result/error
 */
const validateDateRange = (
  startDateStr,
  endDateStr,
  maxDuration,
  allowFutureDate = false
) => {
  let errorMessage, startDate, endDate;

  if (!startDateStr || !endDateStr) {
    errorMessage = 'must include both start date and end date';
  } else {
    try {
      startDate = convertAndValidateDate(startDateStr, 'start date');
      endDate = convertAndValidateDate(endDateStr, 'end date');
    } catch (e) {
      errorMessage = e.message;
    }
    const duration = endDate - startDate;
    if (duration < 0) {
      errorMessage = 'end date cannot be earlier than start date';
    } else if (maxDuration && duration > maxDuration) {
      // TODO: write a formator that uses the most sensible time unit to describe duration.
      errorMessage = `start date and end date cannot be more than ${maxDuration} apart`;
    } else if (!allowFutureDate && endDate - new Date() > 0) {
      errorMessage = 'end date must be no later than today';
    }
  }
  return {
    isValid: !errorMessage,
    error: errorMessage,
    startDate,
    endDate,
  };
};

export { convertAndValidateDate, validateDateRange };
