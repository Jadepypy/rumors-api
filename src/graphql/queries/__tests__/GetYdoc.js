import gql from 'util/GraphQL';
import { loadFixtures, unloadFixtures } from 'util/fixtures';
import fixtures from '../__fixtures__/GetYdoc';

describe('GetYdoc', () => {
  beforeAll(() => loadFixtures(fixtures));
  afterAll(() => unloadFixtures(fixtures));

  it('should get the specified doc', async () => {
    expect(
      await gql`
        {
          GetYdoc(id: "foo") {
            data
            versions {
              createdAt
              snapshot
            }
          }
        }
      `({}, { user: { id: 'test', appId: 'test' } })
    ).toMatchSnapshot();
  });

  it('should return empty versions when there is none', async () => {
    expect(
      await gql`
        {
          GetYdoc(id: "foo2") {
            data
            versions {
              createdAt
              snapshot
            }
          }
        }
      `({}, { user: { id: 'test', appId: 'test' } })
    ).toMatchSnapshot();
  });
});
