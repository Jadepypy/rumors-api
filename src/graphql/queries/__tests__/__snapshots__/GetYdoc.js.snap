// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetYdoc should get the specified doc 1`] = `
Object {
  "data": Object {
    "GetYdoc": Object {
      "data": "B0Oav/LNDwAnAQV1c2VyczB7ImlkIjoiQVdBX1BiT2J5Q2RTLW5XaHVrcS0iLCJuYW1lIjoi6YKK57ej5Lq6In0BJwCav/LNDwADaWRzACcAmr/yzQ8AAmRzAAgAmr/yzQ8BBXtB7zeT80AAAHtB7zeT80AAAHtB7zeT80AAAHtB7zeT80AAAHtB7zeT80AAAAcBC3Byb3NlbWlycm9yAwlwYXJhZ3JhcGgHAJq/8s0PCAYEAJq/8s0PCQF0iJq/8s0PBwJ7Qe83k/NAAAB7Qe83k/NAAACEmr/yzQ8KAWWImr/yzQ8MAXtB7zeT80AAAISav/LNDw0Bc4iav/LNDw4Be0HvN5PzQAAAhJq/8s0PDwF0iJq/8s0PEAF7Qe83k/NAAACEmr/yzQ8RASCImr/yzQ8SAXtB7zeT80AAAISav/LNDxMBbYiav/LNDxQBe0HvN5PzQAAAhJq/8s0PFQFliJq/8s0PFgF7Qe83k/NAAACEmr/yzQ8XAXOImr/yzQ8YAXtB7zeT80AAAISav/LNDxkBZ4iav/LNDxoBe0HvN5PzQAAAAwCav/LNDwIJAZq/8s0PARsBg5q/8s0PHQkBmr/yzQ8BGwGDmr/yzQ8eCQGav/LNDwEbAYOav/LNDx8JAZq/8s0PARsBg5q/8s0PIAkBmr/yzQ8BGwGDmr/yzQ8hCQGav/LNDwEbAYOav/LNDyIJAZq/8s0PARsBg5q/8s0PIwkBmr/yzQ8BGwGDmr/yzQ8kCQGav/LNDwEbAYOav/LNDyUJAZq/8s0PARsBg5q/8s0PJgkBmr/yzQ8BGwGDmr/yzQ8nCQGav/LNDwEbAYOav/LNDygJAZq/8s0PARsBg5q/8s0PKQkBmr/yzQ8BGwGDmr/yzQ8qCQGav/LNDwEbAYiav/LNDxwBe0HvN5PzQAAAg5q/8s0PKwkBmr/yzQ8BGQGDmr/yzQ8tCQGav/LNDwEZAYOav/LNDy4JAZq/8s0PARkBg5q/8s0PLwkBmr/yzQ8BGQGDmr/yzQ8wCQGav/LNDwEZAYOav/LNDzEJAZq/8s0PARkBg5q/8s0PMgkBmr/yzQ8BGQGDmr/yzQ8zCQGav/LNDwEZAYOav/LNDzQJAZq/8s0PARkBg5q/8s0PNQkBmr/yzQ8BGQGDmr/yzQ82CQGav/LNDwEZAYOav/LNDzcJAZq/8s0PARkBg5q/8s0POAkBmr/yzQ8BGQGDmr/yzQ85CQGav/LNDwEZAYOav/LNDzoJAZq/8s0PARkBg5q/8s0POwkBmr/yzQ8BGQGImr/yzQ8sAXtB7zeT80AAAISav/LNDxsBc4iav/LNDz0Be0HvN5PzQAAAhJq/8s0PPgFziJq/8s0PPwF7Qe83k/NAAACEmr/yzQ9AAWGImr/yzQ9BAXtB7zeT80AAAISav/LND0IBZ4iav/LND0MBe0HvN5PzQAAAhJq/8s0PRAFliJq/8s0PRQF7Qe83k/NAAAABmM7mmA0AiPKRqOEMcQN7QepjNOMAAAB7QepjNOMAAAB7QepjNOMAAABV8pGo4QwAiNTKkPALAgl7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAACHmr/yzQ8IAwlwYXJhZ3JhcGiI8pGo4QwIAntB6YVBHkAAAHtB6YVBHkAAAAcA8pGo4QwJBgQA8pGo4QwMDHRlc3QgbWVzc2FnZYjykajhDAsCe0HphUEeQAAAe0HphUEeQAAAhPKRqOEMGAEgiPKRqOEMGgN7QemFQR5AAAB7QemFQR5AAAB7QemFQR5AAABE8pGo4QwNAV+I8pGo4QweA3tB6YVBHkAAAHtB6YVBHkAAAHtB6YVBHkAAAETykajhDB8Bb4jykajhDCIDe0HphUEeQAAAe0HphUEeQAAAe0HphUEeQAAAg5q/8s0PPAkB8pGo4QwBGwGD8pGo4QwnCQHykajhDAEbAYPykajhDCgJAfKRqOEMARsBg/KRqOEMKQkB8pGo4QwBGwGD8pGo4QwqCQHykajhDAEbAYPykajhDCsJAfKRqOEMARsBg/KRqOEMLAkB8pGo4QwBGwGD8pGo4QwtCQHykajhDAEbAYPykajhDC4JAfKRqOEMARsBg/KRqOEMLwkB8pGo4QwBGwGD8pGo4QwwCQHykajhDAEbAYPykajhDDEJAfKRqOEMARsBg/KRqOEMMgkB8pGo4QwBGwGD8pGo4QwzCQHykajhDAEbAYPykajhDDQJAfKRqOEMARsBg/KRqOEMNQkB8pGo4QwBGwGD8pGo4Qw2CQHykajhDAEbAYPykajhDDcJAfKRqOEMARsBg/KRqOEMOAkB8pGo4QwBGwGD8pGo4Qw5CQHykajhDAEbAYPykajhDDoJAfKRqOEMARsBg/KRqOEMOwkB8pGo4QwBGwGI8pGo4QwmAXtB6YVBHkAAAITykajhDBsBX4jykajhDD0Be0HphUEeQAAAhPKRqOEMPgEwiPKRqOEMPwF7QemFQR5AAACH8pGo4QwJAwlwYXJhZ3JhcGiI8pGo4QxBAntB6YVBHkAAAHtB6YVBHkAAAAcA8pGo4QxCBgQA8pGo4QxFAXGI8pGo4QxEAntB6YVBHkAAAHtB6YVBHkAAAITykajhDEYBcYjykajhDEgBe0HphUEeQAAAhPKRqOEMSQFwiPKRqOEMSgF7QemFQR5AAACE8pGo4QxLAWWI8pGo4QxMAXtB6YVBHkAAAIPykajhDDwJAfKRqOEMAU0Bg/KRqOEMTwkB8pGo4QwBTQGD8pGo4QxQCQHykajhDAFNAYPykajhDFEJAfKRqOEMAU0Bg/KRqOEMUgkB8pGo4QwBTQGD8pGo4QxTCQHykajhDAFNAYPykajhDFQJAfKRqOEMAU0Bg/KRqOEMVQkB8pGo4QwBTQGD8pGo4QxWCQHykajhDAFNAYPykajhDFcJAfKRqOEMAU0Bg/KRqOEMWAkB8pGo4QwBTQGD8pGo4QxZCQHykajhDAFNAYPykajhDFoJAfKRqOEMAU0Bg/KRqOEMWwkB8pGo4QwBTQGD8pGo4QxcCQHykajhDAFNAYPykajhDF0JAfKRqOEMAU0Bg/KRqOEMXgkB8pGo4QwBTQGD8pGo4QxfCQHykajhDAFNAYPykajhDGAJAfKRqOEMAU0Bg/KRqOEMYQkB8pGo4QwBTQGD8pGo4QxiCQHykajhDAFNAYPykajhDGMJAfKRqOEMAU0Bg/KRqOEMZAkB8pGo4QwBTQGD8pGo4QxlCQHykajhDAFNAYPykajhDGYJAfKRqOEMAU0Bg/KRqOEMZwkB8pGo4QwBTQGD8pGo4QxoCQHykajhDAFNAYPykajhDGkJAfKRqOEMAU0Bg/KRqOEMagkB8pGo4QwBTQGD8pGo4QxrCQHykajhDAFNAYPykajhDGwJAfKRqOEMAU0Bg/KRqOEMbQkB8pGo4QwBTQGI8pGo4QxOAXtB6YVBHkAAAITykajhDE0BcojykajhDG8Be0HphUEeQAAAAdTKkPALAIj15pHQBAIDe0HnwISqgAAAe0HnwISqgAAAe0HnwISqgAAAB9SnsKEIAIjVpIP8A5IDB3tB4IWCeoAAAHtB4IWCeoAAAHtB4IWCeoAAAHtB4IWCeoAAAHtB4IWCeoAAAHtB4IWCeoAAAHtB4IWCeoAAAITykajhDHABMojUp7ChCAYBe0HghYJ6gAAAhNSnsKEIBwEyiNSnsKEICAF7QeCFgnqAAACE1KewoQgJATOI1KewoQgKAXtB4IWCeoAAAAH15pHQBACImr/yzQ9HA321zaOgCX21zaOgCX21zaOgCZ8C1aSD/AMAiJjO5pgNAgl9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+AeD8pGo4QxuEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMJEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMKEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMLEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMMEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMNEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMOEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMPEwHykajhDAYJAQwNHwEjAT4BQAGD1aSD/AMQEwHykajhDAYJAQwNHwEjAT4BQAGI1aSD/AMIBH2VyYb4B32VyYb4B32VyYb4B32VyYb4B4fykajhDEIDCXBhcmFncmFwaIjVpIP8AxUCfZXJhvgHfZXJhvgHBwDVpIP8AxYGBADVpIP8AxkBYYjVpIP8AxgCfZXJhvgHfZXJhvgHhNWkg/wDGgFhiNWkg/wDHAF9lcmG+AeD1aSD/AMRCQHVpIP8AwEdAYPVpIP8Ax8JAdWkg/wDAR0Bg9Wkg/wDIAkB1aSD/AMBHQGD1aSD/AMhCQHVpIP8AwEdAYPVpIP8AyIJAdWkg/wDAR0Bg9Wkg/wDIwkB1aSD/AMBHQGD1aSD/AMkCQHVpIP8AwEdAYPVpIP8AyUJAdWkg/wDAR0Bg9Wkg/wDJgkB1aSD/AMBHQGD1aSD/AMnCQHVpIP8AwEdAYPVpIP8AygJAdWkg/wDAR0Bg9Wkg/wDKQkB1aSD/AMBHQGD1aSD/AMqCQHVpIP8AwEdAYPVpIP8AysJAdWkg/wDAR0Bg9Wkg/wDLAkB1aSD/AMBHQGD1aSD/AMtCQHVpIP8AwEdAYPVpIP8Ay4JAdWkg/wDAR0Bg9Wkg/wDLwkB1aSD/AMBHQGI1aSD/AMeAX2VyYb4B4PVpIP8AzAJAdWkg/wDARoBg9Wkg/wDMgkB1aSD/AMBGgGD1aSD/AMzCQHVpIP8AwEaAYPVpIP8AzQJAdWkg/wDARoBg9Wkg/wDNQkB1aSD/AMBGgGD1aSD/AM2CQHVpIP8AwEaAYPVpIP8AzcJAdWkg/wDARoBg9Wkg/wDOAkB1aSD/AMBGgGD1aSD/AM5CQHVpIP8AwEaAYPVpIP8AzoJAdWkg/wDARoBg9Wkg/wDOwkB1aSD/AMBGgGD1aSD/AM8CQHVpIP8AwEaAYPVpIP8Az0JAdWkg/wDARoBg9Wkg/wDPgkB1aSD/AMBGgGD1aSD/AM/CQHVpIP8AwEaAYPVpIP8A0AJAdWkg/wDARoBg9Wkg/wDQQkB1aSD/AMBGgGD1aSD/ANCCQHVpIP8AwEaAYPVpIP8A0MJAdWkg/wDARoBiNWkg/wDMQF9lcmG+AeE1aSD/AMdQEFwcS84czBQQWhrQkd3SHlrYWpoREFJYkFVMEJCSnEvOHMwUFNQS1JxT0VNY3RUS2tQQUxBL1hta2RBRUF3PT2I1aSD/ANFAX2VyYb4B4PVpIP8A0QJAdWkg/wDAUZAg9Wkg/wDhwEJAdWkg/wDAUZAg9Wkg/wDiAEJAdWkg/wDAUZAg9Wkg/wDiQEJAdWkg/wDAUZAg9Wkg/wDigEJAdWkg/wDAUZAg9Wkg/wDiwEJAdWkg/wDAUZAg9Wkg/wDjAEJAdWkg/wDAUZAg9Wkg/wDjQEJAdWkg/wDAUZAg9Wkg/wDjgEJAdWkg/wDAUZAg9Wkg/wDjwEJAdWkg/wDAUZAg9Wkg/wDkAEJAdWkg/wDAUZAg9Wkg/wDkQEJAdWkg/wDAUZAg9Wkg/wDkgEJAdWkg/wDAUZAg9Wkg/wDkwEJAdWkg/wDAUZAg9Wkg/wDlAEJAdWkg/wDAUZAg9Wkg/wDlQEJAdWkg/wDAUZAg9Wkg/wDlgEJAdWkg/wDAUZAg9Wkg/wDlwEJAdWkg/wDAUZAg9Wkg/wDmAEJAdWkg/wDAUZAg9Wkg/wDmQEJAdWkg/wDAUZAg9Wkg/wDmgEJAdWkg/wDAUZARNWkg/wDGgFhxNWkg/wDGtWkg/wDHQFhiNWkg/wDhgECfZXJhvgHfZXJhvgHg9Wkg/wDmwEKAdWkg/wDAZ0BAYPVpIP8A6ABCgHVpIP8AwGdAQGD1aSD/AOhAQoB1aSD/AMBnQEBg9Wkg/wDogEKAdWkg/wDAZ0BAYPVpIP8A6MBCgHVpIP8AwGdAQGD1aSD/AOkAQoB1aSD/AMBnQEBg9Wkg/wDpQEKAdWkg/wDAZ0BAYPVpIP8A6YBCgHVpIP8AwGdAQGD1aSD/AOnAQoB1aSD/AMBnQEBg9Wkg/wDqAEKAdWkg/wDAZ0BAYPVpIP8A6kBCgHVpIP8AwGdAQGD1aSD/AOqAQoB1aSD/AMBnQEBg9Wkg/wDqwEKAdWkg/wDAZ0BAYPVpIP8A6wBCgHVpIP8AwGdAQGD1aSD/AOtAQoB1aSD/AMBnQEBg9Wkg/wDrgEKAdWkg/wDAZ0BAYPVpIP8A68BCgHVpIP8AwGdAQGD1aSD/AOwAQoB1aSD/AMBnQEBg9Wkg/wDsQEKAdWkg/wDAZ0BAYPVpIP8A7IBCgHVpIP8AwGdAQGD1aSD/AOzAQoB1aSD/AMBnQEBg9Wkg/wDtAEKAdWkg/wDAZ0BAYPVpIP8A7UBCgHVpIP8AwGdAQGI1aSD/AOfAQJ9lcmG+Ad9lcmG+AeD1aSD/AO2AQwB1aSD/AMCGQGcAQGD1aSD/AO5AQwB1aSD/AMCGQGcAQGD1aSD/AO6AQwB1aSD/AMCGQGcAQGD1aSD/AO7AQwB1aSD/AMCGQGcAQGD1aSD/AO8AQwB1aSD/AMCGQGcAQGD1aSD/AO9AQwB1aSD/AMCGQGcAQGD1aSD/AO+AQwB1aSD/AMCGQGcAQGD1aSD/AO/AQwB1aSD/AMCGQGcAQGD1aSD/APAAQwB1aSD/AMCGQGcAQGD1aSD/APBAQwB1aSD/AMCGQGcAQGD1aSD/APCAQwB1aSD/AMCGQGcAQGD1aSD/APDAQwB1aSD/AMCGQGcAQGD1aSD/APEAQwB1aSD/AMCGQGcAQGD1aSD/APFAQwB1aSD/AMCGQGcAQGD1aSD/APGAQwB1aSD/AMCGQGcAQGD1aSD/APHAQwB1aSD/AMCGQGcAQGD1aSD/APIAQwB1aSD/AMCGQGcAQGD1aSD/APJAQwB1aSD/AMCGQGcAQGD1aSD/APKAQwB1aSD/AMCGQGcAQGD1aSD/APLAQwB1aSD/AMCGQGcAQGD1aSD/APMAQwB1aSD/AMCGQGcAQGD1aSD/APNAQwB1aSD/AMCGQGcAQGD1aSD/APOAQwB1aSD/AMCGQGcAQGD1aSD/APPAQwB1aSD/AMCGQGcAQGI1aSD/AO4AQJ9lcmG+Ad9lcmG+AeD1aSD/APQAQkB1aSD/AMBFgGD1aSD/APTAQkB1aSD/AMBFgGD1aSD/APUAQkB1aSD/AMBFgGD1aSD/APVAQkB1aSD/AMBFgGD1aSD/APWAQkB1aSD/AMBFgGD1aSD/APXAQkB1aSD/AMBFgGD1aSD/APYAQkB1aSD/AMBFgGD1aSD/APZAQkB1aSD/AMBFgGD1aSD/APaAQkB1aSD/AMBFgGD1aSD/APbAQkB1aSD/AMBFgGD1aSD/APcAQkB1aSD/AMBFgGD1aSD/APdAQkB1aSD/AMBFgGD1aSD/APeAQkB1aSD/AMBFgGD1aSD/APfAQkB1aSD/AMBFgGD1aSD/APgAQkB1aSD/AMBFgGD1aSD/APhAQkB1aSD/AMBFgGD1aSD/APiAQkB1aSD/AMBFgGD1aSD/APjAQkB1aSD/AMBFgGD1aSD/APkAQkB1aSD/AMBFgGD1aSD/APlAQkB1aSD/AMBFgGD1aSD/APmAQkB1aSD/AMBFgGD1aSD/APnAQkB1aSD/AMBFgGD1aSD/APoAQkB1aSD/AMBFgGD1aSD/APpAQkB1aSD/AMBFgGD1aSD/APqAQkB1aSD/AMBFgGD1aSD/APrAQkB1aSD/AMBFgHHmr/yzQ8I8pGo4QwJAwlwYXJhZ3JhcGgHANWkg/wD7QEGBADVpIP8A+4BDHRlc3QgbWVzc2FnZUTVpIP8A+8BAV9E1aSD/AP7AQFvhNWkg/wD+gECXzCI1aSD/APSAQZ9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+Ad9lcmG+AeD1aSD/APsAQoB1aSD/AMB7wEQg9Wkg/wDhQIKAdWkg/wDAe8BEIPVpIP8A4YCCgHVpIP8AwHvARCD1aSD/AOHAgoB1aSD/AMB7wEQg9Wkg/wDiAIKAdWkg/wDAe8BEIPVpIP8A4kCCgHVpIP8AwHvARCD1aSD/AOKAgoB1aSD/AMB7wEQg9Wkg/wDiwIKAdWkg/wDAe8BEIPVpIP8A4wCCgHVpIP8AwHvARCD1aSD/AONAgoB1aSD/AMB7wEQg9Wkg/wDjgIKAdWkg/wDAe8BEIPVpIP8A48CCgHVpIP8AwHvARCD1aSD/AOQAgoB1aSD/AMB7wEQg9Wkg/wDkQIKAdWkg/wDAe8BEIPVpIP8A5ICCgHVpIP8AwHvARCD1aSD/AOTAgoB1aSD/AMB7wEQg9Wkg/wDlAIKAdWkg/wDAe8BEIPVpIP8A5UCCgHVpIP8AwHvARCD1aSD/AOWAgoB1aSD/AMB7wEQg9Wkg/wDlwIKAdWkg/wDAe8BEIPVpIP8A5gCCgHVpIP8AwHvARCD1aSD/AOZAgoB1aSD/AMB7wEQg9Wkg/wDmgIKAdWkg/wDAe8BEIPVpIP8A5sCCgHVpIP8AwHvARCD1aSD/AOcAgoB1aSD/AMB7wEQg9Wkg/wDnQIKAdWkg/wDAe8BEIPVpIP8A54CCgHVpIP8AwHvARCD1aSD/AOfAgoB1aSD/AMB7wEQg9Wkg/wDoAIKAdWkg/wDAe8BEIPVpIP8A6ECCgHVpIP8AwHvARCD1aSD/AOiAgoB1aSD/AMB7wEQg9Wkg/wDowIKAdWkg/wDAe8BEIPVpIP8A6QCCgHVpIP8AwHvARCI1aSD/AOEAgJ9lcmG+Ad9lcmG+AeD1aSD/AOlAgoB1aSD/AMB7QECg9Wkg/wDqAIKAdWkg/wDAe0BAoPVpIP8A6kCCgHVpIP8AwHtAQKD1aSD/AOqAgoB1aSD/AMB7QECg9Wkg/wDqwIKAdWkg/wDAe0BAoPVpIP8A6wCCgHVpIP8AwHtAQKD1aSD/AOtAgoB1aSD/AMB7QECg9Wkg/wDrgIKAdWkg/wDAe0BAoPVpIP8A68CCgHVpIP8AwHtAQKD1aSD/AOwAgoB1aSD/AMB7QECg9Wkg/wDsQIKAdWkg/wDAe0BAoPVpIP8A7ICCgHVpIP8AwHtAQKD1aSD/AOzAgoB1aSD/AMB7QECg9Wkg/wDtAIKAdWkg/wDAe0BAoPVpIP8A7UCCgHVpIP8AwHtAQKD1aSD/AO2AgoB1aSD/AMB7QECg9Wkg/wDtwIKAdWkg/wDAe0BAoPVpIP8A7gCCgHVpIP8AwHtAQKD1aSD/AO5AgoB1aSD/AMB7QECg9Wkg/wDugIKAdWkg/wDAe0BAoPVpIP8A7sCCgHVpIP8AwHtAQKD1aSD/AO8AgoB1aSD/AMB7QECg9Wkg/wDvQIKAdWkg/wDAe0BAoPVpIP8A74CCgHVpIP8AwHtAQKD1aSD/AO/AgoB1aSD/AMB7QECg9Wkg/wDwAIKAdWkg/wDAe0BAoPVpIP8A8ECCgHVpIP8AwHtAQKD1aSD/APCAgoB1aSD/AMB7QECg9Wkg/wDwwIKAdWkg/wDAe0BAoPVpIP8A8QCCgHVpIP8AwHtAQKD1aSD/APFAgoB1aSD/AMB7QECg9Wkg/wDxgIKAdWkg/wDAe0BAoPVpIP8A8cCCgHVpIP8AwHtAQKD1aSD/APIAgoB1aSD/AMB7QECg9Wkg/wDyQIKAdWkg/wDAe0BAojVpIP8A6cCBH2VyYb4B32VyYb4B32VyYb4B32VyYb4B8fykajhDELVpIP8AxYDCXBhcmFncmFwaIjVpIP8A84CAn2VyYb4B32VyYb4BwcA1aSD/APPAgYEANWkg/wD0gIPb190ZXN0IG1lc3NhZ2VfhNWkg/wD4QIBMIjVpIP8A9ECAn2VyYb4B32VyYb4B4PVpIP8A8oCCgHVpIP8AwHiAgGD1aSD/APlAgoB1aSD/AMB4gIBg9Wkg/wD5gIKAdWkg/wDAeICAYPVpIP8A+cCCgHVpIP8AwHiAgGD1aSD/APoAgoB1aSD/AMB4gIBg9Wkg/wD6QIKAdWkg/wDAeICAYPVpIP8A+oCCgHVpIP8AwHiAgGD1aSD/APrAgoB1aSD/AMB4gIBg9Wkg/wD7AIKAdWkg/wDAeICAYPVpIP8A+0CCgHVpIP8AwHiAgGD1aSD/APuAgoB1aSD/AMB4gIBg9Wkg/wD7wIKAdWkg/wDAeICAYPVpIP8A/ACCgHVpIP8AwHiAgGD1aSD/APxAgoB1aSD/AMB4gIBg9Wkg/wD8gIKAdWkg/wDAeICAYPVpIP8A/MCCgHVpIP8AwHiAgGD1aSD/AP0AgoB1aSD/AMB4gIBg9Wkg/wD9QIKAdWkg/wDAeICAYPVpIP8A/YCCgHVpIP8AwHiAgGD1aSD/AP3AgoB1aSD/AMB4gIBg9Wkg/wD+AIKAdWkg/wDAeICAYPVpIP8A/kCCgHVpIP8AwHiAgGD1aSD/AP6AgoB1aSD/AMB4gIBg9Wkg/wD+wIKAdWkg/wDAeICAYPVpIP8A/wCCgHVpIP8AwHiAgGD1aSD/AP9AgoB1aSD/AMB4gIBg9Wkg/wD/gIKAdWkg/wDAeICAYPVpIP8A/8CCgHVpIP8AwHiAgGD1aSD/AOAAwoB1aSD/AMB4gIBg9Wkg/wDgQMKAdWkg/wDAeICAYPVpIP8A4IDCgHVpIP8AwHiAgGD1aSD/AODAwoB1aSD/AMB4gIBg9Wkg/wDhAMKAdWkg/wDAeICAYPVpIP8A4UDCgHVpIP8AwHiAgGD1aSD/AOGAwoB1aSD/AMB4gIBg9Wkg/wDhwMKAdWkg/wDAeICAYPVpIP8A4gDCgHVpIP8AwHiAgGD1aSD/AOJAwoB1aSD/AMB4gIBg9Wkg/wDigMKAdWkg/wDAeICAYPVpIP8A4sDCgHVpIP8AwHiAgGD1aSD/AOMAwoB1aSD/AMB4gIBg9Wkg/wDjQMKAdWkg/wDAeICAYPVpIP8A44DCgHVpIP8AwHiAgGI1aSD/APkAgF9lcmG+AeE1aSD/APiAgExiNWkg/wDkAMBfZXJhvgHA5q/8s0PAhkBGwHykajhDAgJAQwNGwEfASMBPgFAAU0B1aSD/AMHFgEZAh0BRkCcAQLtARLiAgE=",
      "versions": Array [
        Object {
          "createdAt": "2023-09-07T08:14:14.005Z",
          "snapshot": "AZq/8s0PAhkBGwEBmr/yzQ9I",
        },
        Object {
          "createdAt": "2023-09-07T08:16:45.613Z",
          "snapshot": "Apq/8s0PAhkBGwHykajhDAIbAU0BBJq/8s0PSPKRqOEMctTKkPALA/XmkdAEAw==",
        },
        Object {
          "createdAt": "2023-09-07T08:18:32.467Z",
          "snapshot": "A5q/8s0PAhkBGwHykajhDAgJAQwNGwEfASMBPgFAAU0B1aSD/AMHFgEZAh0BRkCcAQLtARLiAgEGmr/yzQ9ImM7mmA0D8pGo4Qxy1MqQ8AsD9eaR0AQD1aSD/AOTAw==",
        },
        Object {
          "createdAt": "2023-09-07T08:18:49.500Z",
          "snapshot": "A5q/8s0PAhkBGwHykajhDAgJAQwNGwEfASMBPgFAAU0B1aSD/AMHFgEZAh0BRkCcAQLtARLiAgEHmr/yzQ9ImM7mmA0D8pGo4Qxy1MqQ8AsD1KewoQgN9eaR0AQD1aSD/AOTAw==",
        },
      ],
    },
  },
}
`;

exports[`GetYdoc should return empty versions when there is none 1`] = `
Object {
  "data": Object {
    "GetYdoc": Object {
      "data": "AgzOmK+vDwAnAQV1c2VyczB7ImlkIjoiQVdBX1BiT2J5Q2RTLW5XaHVrcS0iLCJuYW1lIjoi6YKK57ej5Lq6In0BJwDOmK+vDwADaWRzACcAzpivrw8AAmRzAAgAzpivrw8BBXtB7r15icAAAHtB7r15icAAAHtB7r15icAAAHtB7r15icAAAHtB7r15icAAAAcBC3Byb3NlbWlycm9yAwlwYXJhZ3JhcGgHAM6Yr68PCAYEAM6Yr68PCQExiM6Yr68PBwJ7Qe69eYnAAAB7Qe69eYnAAACEzpivrw8KATKIzpivrw8MAXtB7r15icAAAITOmK+vDw0BM4jOmK+vDw4Be0HuvXmJwAAACvmO4JABAIjOmK+vDxAFfbmdwKECfbmdwKECfbmdwKECfbmdwKECfbmdwKECh86Yr68PCAMJcGFyYWdyYXBoiPmO4JABBAJ9uZ3AoQJ9uZ3AoQIHAPmO4JABBQYEAPmO4JABCAE0iPmO4JABBwJ9uZ3AoQJ9uZ3AoQKE+Y7gkAEJATWI+Y7gkAELAX25ncChAoT5juCQAQwBNoj5juCQAQ0BfbmdwKECAA==",
      "versions": Array [],
    },
  },
}
`;
